src/main/java/io/stargate/sgv2/jsonapi/service/operation/MetadataAttemptPage.java

```diff
@@ -0,0 +1,74 @@
+package io.stargate.sgv2.jsonapi.service.operation;
+
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResultBuilder;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.KeyspaceSchemaObject;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import java.util.function.Supplier;
+
+/**
+ * A page of results for metadata based command, use {@link #builder()} to get a builder to pass to
+ * {@link GenericOperation}.
+ */
+public abstract class MetadataAttemptPage<SchemaT extends SchemaObject>
+    extends OperationAttemptPage<SchemaT, MetadataAttempt<SchemaT>> {
+
+  private final boolean showSchema;
+  private final CommandStatus statusKey;
+
+  private MetadataAttemptPage(
+      OperationAttemptContainer<SchemaT, MetadataAttempt<SchemaT>> attempts,
+      CommandResultBuilder resultBuilder,
+      boolean showSchema,
+      CommandStatus statusKey) {
+    super(attempts, resultBuilder);
+    this.showSchema = showSchema;
+    this.statusKey = statusKey;
+  }
+
+  public static <SchemaT extends KeyspaceSchemaObject>
+      MetadataAttemptPage.Builder<SchemaT> builder() {
+    return new MetadataAttemptPage.Builder<>();
+  }
+
+  @Override
+  protected void buildCommandResult() {
+    addAttemptWarningsToResult();
+    addAttemptErrorsToResult();
+    if (attempts.get(0).status() != OperationAttempt.OperationStatus.ERROR) {
```

Comment:
attempts is a OperationAttemptContainer and has completedAttempts() to get completed attempts , it checks actually COMPLETED not just not ERROR

we should prob also check if all the size of attempts is != 1 and throw a regular InvalidStateExcpetion as a double check.